server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java

```diff
@@ -0,0 +1,81 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.metrics.dropwizard;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+
+public class ESReporterConfiguration {
+
+    public static final boolean ENABLED = true;
+    public static final boolean DISABLED = !ENABLED;
+    public static final String DEFAULT_INDEX = "james-metrics";
+    public static final long DEFAULT_PERIOD_IN_SECOND = 60L;
+
+    public static ESReporterConfiguration disabled() {
+        return new ESReporterConfiguration(
+            Optional.<String>absent(),
+            Optional.<Integer>absent(),
+            DISABLED,
+            Optional.<String>absent(),
+            Optional.<Long>absent());
+    }
+
+    public static ESReporterConfiguration enabled(String host, int port, Optional<String> index, Optional<Long> periodInSecond) {
+        return new ESReporterConfiguration(
+            Optional.of(host),
+            Optional.of(port),
+            ENABLED,
+            index,
+            periodInSecond);
+    }
```

Comment:
can you make a builder? There are too much basic misleading types in this constructor (int/long and 2 strings)